go/enclave/rollupchain/rollup_chain.go

```diff
@@ -469,21 +438,16 @@ func allReceipts(txReceipts []*types.Receipt, depositReceipts []*types.Receipt)
 }
 
 // SubmitBlock is used to update the enclave with an additional L1 block.
-func (rc *RollupChain) SubmitBlock(block types.Block) common.BlockSubmissionResponse {
+func (rc *RollupChain) SubmitBlock(block types.Block, isLatest bool) common.BlockSubmissionResponse {
 	rc.blockProcessingMutex.Lock()
 	defer rc.blockProcessingMutex.Unlock()
 
-	// The genesis block should always be ingested, not submitted, so we ignore it if it's passed in here.
-	if rc.isGenesisBlock(&block) {
-		return common.BlockSubmissionResponse{IngestedBlock: false, BlockNotIngestedCause: "Block was genesis block."}
-	}
-
 	_, foundBlock := rc.storage.FetchBlock(block.Hash())
 	if foundBlock {
 		return common.BlockSubmissionResponse{IngestedBlock: false, BlockNotIngestedCause: "Block already ingested."}
 	}
 
-	if ingestionFailedResponse := rc.insertBlockIntoL1Chain(&block); ingestionFailedResponse != nil {
+	if ingestionFailedResponse := rc.insertBlockIntoL1Chain(&block); !rc.isGenesisBlock(&block) && ingestionFailedResponse != nil {
```

Comment:
This check was being made by the `Ingest` method, inserting the genesis block fails but we still need to store it so we don't want to error out here.

This will go away when I make the change to start reading from the management contract deploy block so not too worried about it being ugly for now.